website/www/site/content/en/documentation/ml/runinference-metrics.md

```diff
@@ -0,0 +1,102 @@
+
+We showcase different RunInference metrics when the pipeline is executed using the Dataflow Runner on CPU and GPU. The full example code can be found [here](https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/inference/runinference_metrics/).
+
+
+The file structure for entire pipeline is:
+
+    runinference_metrics/
+    ├── pipeline/
+    │   ├── __init__.py
+    │   ├── options.py
+    │   └── transformations.py
+    ├── __init__.py
+    ├── config.py
+    ├── main.py
+    └── setup.py
+
+`pipeline/transformations.py` contains the code for `beam.DoFn` and additional functions that are used for pipeline
+
+`pipeline/options.py` contains the pipeline options to configure the Dataflow pipeline
+
+`config.py` defines some variables like GCP `PROJECT_ID`, `NUM_WORKERS` that are used multiple times
+
+`setup.py` defines the packages/requirements for the pipeline to run
+
+`main.py` contains the pipeline code and some additional functions used for running the pipeline
+
+
+### How to Run the Pipeline
+First, make sure you have installed the required packages. One should have access to a Google Cloud Project and then correctly configure the GCP variables like `PROJECT_ID`, `REGION`, and others in `config.py`. To use GPUs, follow the setup instructions [here](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/dataflow/gpu-examples/pytorch-minimal).
+
+
+1. Dataflow with CPU: `python main.py --mode cloud --device CPU`
+2. Dataflow with GPU: `python main.py --mode cloud --device GPU`
+
+The pipeline can be broken down into few simple steps:
+1. Create a list of texts to use as an input using `beam.Create`
+2. Tokenize the text
+3. Use RunInference to do inference
+4. Postprocess the output of RunInference
+
+{{< highlight >}}
+  with beam.Pipeline(options=pipeline_options) as pipeline:
+    _ = (
+        pipeline
+        | "Create inputs" >> beam.Create(inputs)
+        | "Tokenize" >> beam.ParDo(Tokenize(cfg.TOKENIZER_NAME))
+        | "Inference" >>
+        RunInference(model_handler=KeyedModelHandler(model_handler))
+        | "Decode Predictions" >> beam.ParDo(PostProcessor()))
+{{< /highlight >}}
+
+
+## RunInference Metrics
+
+As mentioned above, we benchmarked the performance of RunInference using Dataflow on both CPU and GPU. These metrics can be seen in the GCP UI and can also be printed using
+
+{{< highlight >}}
+metrics = pipeline.result.metrics().query(beam.metrics.MetricsFilter())
+{{< /highlight >}}
+
+
+A snapshot of different metrics from GCP UI when using Dataflow on GPU:
+
+  ![runinference-GPU-metrics.png](https://github.com/apache/beam/tree/master/website/www/site/static/images/runinference_metrics_snapshot.svg)
```

Comment:
```suggestion
  ![RunInference GPU metrics rendered on Dataflow](/images/runinference_metrics_snapshot.svg)
```

Right now, this doesn't render correctly - http://apache-beam-website-pull-requests.storage.googleapis.com/23554/documentation/ml/runinference-metrics/index.html#runinference-metrics